File: poc-kronos-frontend/src/model/ocorrencia/Ocorrencia.js
class Ocorrencia {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.data = dados ? dados.data : null;
    this.hora = dados ? dados.hora : null;
    this.registro = dados ? dados.registro : null;
    this.idTurma = dados ? dados.idTurma : null;
    this.idTipoOcorrencia = dados ? dados.idTipoOcorrencia : null;
    this.idMatricula = dados ? dados.idMatricula : null;
    this.idFuncionarioRelator = dados ? dados.idFuncionarioRelator : null;
    this.idFuncionarioRegistro = dados ? dados.idFuncionarioRegistro : null;
    this.dataCiencia = dados ? dados.dataCiencia : null;
    this.idResponsavelCiencia = dados ? dados.idResponsavelCiencia : null;
    this.anulado = dados ? dados.anulado : null;
    this.dataAnulacao = dados ? dados.dataAnulacao : null;
    this.motivoAnulacao = dados ? dados.motivoAnulacao : null;
    this.idFuncionarioAnulacao = dados ? dados.idFuncionarioAnulacao : null;
    this.fator = dados ? dados.fator : null;
    this.nomeFator = dados ? dados.nomeFator : null;
    this.valor = dados ? dados.valor : null;
    this.idPessoaSelecionada = null;
  }

  static validations(required, dataFimMenor) {
    return {
      data: { required },
      hora: { required },
      registro: { required },
      idTurma: { required },
      idTipoOcorrencia: { required },
      idMatricula: { required },
      idFuncionarioRelator: { required },
      dataAnulacao: { dataFimMenor: dataFimMenor("data") },
      dataCiencia: { dataFimMenor: dataFimMenor("data") }
    };
  }

  static validationMessages() {
    return {
      data: {
        required: "Data é obrigatória"
      },
      hora: {
        required: "Hora é obrigatório"
      },
      registro: {
        required: "Registro é obrigatório"
      },
      idTurma: {
        required: "Turma é obrigatório"
      },
      idTipoOcorrencia: {
        required: "Tipo Ocorrência é obrigatório"
      },
      idMatricula: {
        required: "Matrícula é obrigatória"
      },
      idFuncionarioRelator: {
        required: "Relator é obrigatório"
      },
      dataAnulacao: {
        dataFimMenor: "Data Anulação não pode ser menor que a Data da Ocorrência"
      },
      dataCiencia: {
        dataFimMenor: "Data Ciência não pode ser menor que a Data da Ocorrência"
      }
    };
  }
}
export default Ocorrencia;

File: src/main/java/br/com/kronos/backend/aplicacao/ocorrencia/Ocorrencia.java

package br.com.kronos.backend.aplicacao.ocorrencia;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import java.time.LocalDate;
import java.time.LocalTime;

import br.com.kronos.backend.aplicacao.validador.Validacoes;
import lombok.Builder;
import lombok.Getter;

@Getter
public class Ocorrencia {
    
    private Long id;
    private LocalDate data; 
    private LocalTime hora;
    private String registro; 
    private Long idTurma;
    private Long idTipoOcorrencia;
    private Long idMatricula;
    private Long idFuncionarioRelator;
    private Long idFuncionarioRegistro;
    private LocalDate dataCiencia;
    private Long idResponsavelCiencia;
    private boolean anulado;
    private LocalDate dataAnulacao; 
    private String motivoAnulacao;
    private Long idFuncionarioAnulacao;

	@Builder
    public Ocorrencia(Long id, LocalDate data, LocalTime hora, String registro, Long idTurma, Long idTipoOcorrencia,  Long idMatricula,
                      Long idFuncionarioRelator, Long idFuncionarioRegistro, LocalDate dataCiencia, Long idResponsavelCiencia, 
                      boolean anulado, LocalDate dataAnulacao, String motivoAnulacao, Long idFuncionarioAnulacao) {

		super();
        this.id = id;
        this.data = exigirNaoNulo(data, "Data"); 
        this.hora = exigirNaoNulo(hora, "Hora"); 
        this.registro = Validacoes.exigirTamanho(registro, "Registro da ocorrência", 1, 500);       
        this.idTurma = exigirNaoNulo(idTurma, "Turma"); 
        this.idTipoOcorrencia = exigirNaoNulo(idTipoOcorrencia, "Tipo de ocorrência"); 
        this.idMatricula = exigirNaoNulo(idMatricula, "Matrícula (aluno)"); 
        this.idFuncionarioRelator = exigirNaoNulo(idFuncionarioRelator, "Funcionário relator"); 
        this.idFuncionarioRegistro = exigirNaoNulo(idFuncionarioRegistro, "Funcionário de registro"); 
        this.dataCiencia = dataCiencia; 
        this.idResponsavelCiencia = idResponsavelCiencia !=null && idResponsavelCiencia > 0 ? idResponsavelCiencia : null; 
        this.anulado = exigirNaoNulo(anulado, "Anulado"); 
        this.dataAnulacao = dataAnulacao; 
        this.motivoAnulacao = motivoAnulacao; 
        this.idFuncionarioAnulacao = idFuncionarioAnulacao !=null && idFuncionarioAnulacao > 0 ? idFuncionarioAnulacao : null; 
      
	}
}